sunpy/visualization/colormaps/color_tables.py

```diff
@@ -376,3 +377,7 @@ def euvi_color_table(wavelength: u.angstrom):
             "Invalid EUVI wavelength. Valid values are "
             "171, 195, 284, 304."
         )
+
+
+def punch_color_table():
+    return cmap_from_rgb_file("PUNCH", "punch.csv")
```

Comment:
I think you also need to modify `cm.py` to add it to the list that get's loaded by matplotlib.